src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -171,12 +177,15 @@ internal override CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
                 var sourceMethod = this.ContainingSymbol as SourceMemberMethodSymbol;
                 if ((object)sourceMethod == null || (object)sourceMethod.SourcePartialDefinition == null)
                 {
-                    lazyAttributesStored = LoadAndValidateAttributes(OneOrMany.Create(this.MergedAttributeDeclarationSyntaxLists), ref _lazyCustomAttributesBag);
+                    lazyAttributesStored = LoadAndValidateAttributes(
```

Comment:
> I don't understand how GetDeclaredSymbol is the problem here.

GetDeclaredSymbol is not a source of this problem. I am just saying that implementation of GetDeclaredSymbol for local functions has a separate problem, it is not supposed to perform any binding. SemanticModel is supposed to do the minimal possible amount of work to perform the task. The task can be performed without binding, therefore we shouldn't bind in order to find the symbol. Difference in behavior between my test and your test exposes the problem, makes it observable.

> Shouldn't I force diagnostics in that code path as well (and then throw them away)? 

I don't think so. The problem is that for local functions (their parameters, type parameters, etc.) we are reusing symbols that are designed for regular declarations. Part of design for them is that they spill all diagnostics eagerly into global declaration bag. That should never be done for local functions, all diagnostics should be channeled through GetDeclarationDiagnostics(bag) method, no matter what is the order in which symbol APIs are accessed.